app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ApplicationsController < ApplicationController
+
+  def index
+    @applications = Application.all
+  end
+  
+  def new
+    @application = Application.new
+  end
+
+  def create
+    @application = Application.new(application_params)
+    if @application.save
+      redirect_to "/applications/#{@application.id}"   
+    else
+      render "new"
+    end
+  end
+
+  def show
+    @application = Application.find(params[:id])
+    @pets = @application.pets
+  end
+
+  def search_pets
```

Comment:
Instead of making a separate action here, another option would be to pass the search term as a query parameter back to the same show page. Since you're rendering the same view here anyways, you could either turn this into a helper method or instead add the following block to your `show` action:
```
if params[:name].present?
    @search_results = @application.search_pets_by_name(params[:name])
end
```